nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -23,15 +25,37 @@ export function setBlockAssembledWebSocketConnection(_ws) {
   ws = _ws;
 }
 
-async function makeBlock(proposer, number = TRANSACTIONS_PER_BLOCK) {
+async function makeBlock(proposer, unprocessed, number = TRANSACTIONS_PER_BLOCK) {
   logger.debug('Block Assembler - about to make a new block');
   // we retrieve un-processed transactions from our local database, relying on
   // the transaction service to keep the database current
-  const transactions = await getMostProfitableTransactions(number);
+  const unprocessedTransactions = await getMostProfitableTransactions(unprocessed);
+  let { blockNumberL2, blockHash } = await getLatestBlockInfo();
+  let latestTree = await getLatestTree();
+  const blockList = [];
+  const transactionsList = [];
+
   // then we make new block objects until we run out of unprocessed
   // transactions
-  const block = await Block.build({ proposer, transactions });
-  return { block, transactions };
+  for (let i = 0; i < unprocessedTransactions.length / number; i++) {
```

Comment:
We can rid of this `number` parameter that was in this function previously to this PR as it's not being used, and use directly TRANSACTIONS_PER_BLOCK. It will be more clear.